singularity/quit.sh

```diff
@@ -7,14 +7,16 @@ set -eo pipefail
 cd /sw/apps/dpdash
 npm stop
 
-# TODO: make this do a graceful exit
+# kill PM2
+./node_modules/pm2/bin/pm2 kill
 
-pkill -f 'pm2'
+# kill supervisord (thus mongo, celery, erlang, rabbitmq)
+echo "Quitting supervisord..."
+pidfile="/data/dpdash/supervisord/supervisord.pid"
+pid=$(<$pidfile)
+kill -SIGQUIT $pid
 
-# TODO: send SIGQUIT to supervisord
-
-pkill -f 'supervisord'
```

Comment:
Does `supervisord` killing ensure killing all supervised processes?

Reply:
Exactly! http://supervisord.org/running.html#signal-handlers

Reply:
Also, by sending this signal to the PID of this DPdash instance's `supervisord` (by grabbing it from `/data/dpdash/supervisord`), we ensure that any other running instance on the same machine will be untouched. Which also goes for all of its supervised processes.

Reply:
Sure, thanks for devising this.